scripts/idl/test_xml_parser.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import io
+
+from typing import Optional, Union, List
+
+try:
+    from idl.matter_idl_types import *
+    from idl.zapxml import ParseSource, ParseXmls
+except:
+    import os
+    import sys
+
+    sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+
+    from idl.matter_idl_types import *
+    from idl.zapxml import ParseSource, ParseXmls
+
+
+def XmlToIdl(what: Union[str, List[str]]) -> Idl:
+    if not isinstance(what, list):
+        what = [what]
+
+    sources = []
+    for idx, txt in enumerate(what):
+        sources.append(ParseSource(source=io.StringIO(txt), name=("Input %d" % (idx + 1))))
+
+    return ParseXmls(sources, include_meta_data=False)
+
+
+class TestXmlParser(unittest.TestCase):
+
+    def testEmptyInput(self):
+        idl = XmlToIdl('<configurator/>')
+        self.assertEqual(idl, Idl())
+
+    def testCluster(self):
```

Comment:
If we're being exhaustive, I'd recommend we instead adopt elements from the test-cluster since it exhaustively verifies most aspects of the Matter data model.

Reply:
More tests would be useful for sure. For now I have the global CI check that says "all our XMLs are parseable" and while that test does not validate contents, at least it checks that we accept all possible inputs.

I will add more tests as we progress. I would prefer focused tests for every element instead of one huge XML test.